src/activity-logger/background/log-page-visit.ts

```diff
@@ -98,8 +99,41 @@ export default class PageVisitLogger {
             // Don't index full-text in this stage
             delete analysisRes.content.fullText
 
+            const isPDF = tab.url.endsWith('.pdf')
+            const PDFFingerprint = null
+            console.log({
+                url: tab.url,
+                isPDF,
+                PDFFingerprint,
+                ...analysisRes,
+            })
+            if (isPDF) {
+                getPDFFingerprint(tab.url).then(async fingerprint => {
```

Comment:
I understand you haven't got the PDF fingerprint implementation yet, although the async flow of this is going to cause issues. `getPDFFingerprint` will run, then it will immediately move onto the following `this._createPage` call, as there's no `await` and that call is not inside the `.then()` block. And you'll end up calling `this._createPage` twice.

Instead you could do something like:

```ts
const fingerprint = isPDF ? await getPDFFingerprint(tab.url) : undefined
```

and remove the entire `.then()`, as the following `this._createPage` call will then take over. `undefined` over `null` here also as I think if it's `null` you'll end up telling IDB to store a `null` value for lots of pages. If it's `undefined`, IDB just skips that field.